### DISCUSSION TITLE
Skip scheduled flows not started in the past

### DISCUSSION THREAD

**Discussion by menteora** [PrefectHQ/prefect#5005] [Q&A] [answered]
Hi, 
I've 10 Flows not started in scheduled time.

![past](https://user-images.githubusercontent.com/3657745/135217664-a6d4ae95-c75c-44ab-a4ed-e0ccac79711f.jpg)

Now agent starts and run all pending flows.

![now](https://user-images.githubusercontent.com/3657745/135217747-92c1deaf-a7ec-4a4f-99f8-ea1448e372d2.jpg)

I want skip pending flows without execute past scheduled flows.
How I can do this?

Thank you very much for help.

**Comment by kvnkho** [comment, marked as answer]
Hi @menteora 

Sorry for the delay. I think what you are asking if that if an agent goes down, and a lot of runs of the same flow get queued, how do we avoid running those and just moving to the next scheduled run?

Prefect Cloud has Automations that let you cancel the flow run if it is late by a certain amount of time. You can read more on that [here](https://docs.prefect.io/orchestration/concepts/automations.html)

**Reply by menteora** [reply]
Thank you very mutch for your responce and you correctly understand my question.

Is it possibile to get scheduled time from python code and skip execution programmatically? 

I think it's a possible solution that works both server and cloud solutions.

**Reply by kvnkho** [reply]
Hey @menteora 

You can use a state handler at the Flow level to make a GraphQL query and see if there are any flows running with the same name and project. If there are, you can just return a success state and exit the flow. 

Docs for state handler here: https://docs.prefect.io/core/concepts/notifications.html#state-handlers

To get scheduled time from a Flow, you can do `prefect.context.get("scheduled_start_time")` as seen here: https://docs.prefect.io/api/latest/utilities/context.html

**Reply by KayJey-Krayen** [reply]
Is there a similar code to do the same , in prefect 2.0  ?
This code dose not seem to be working in prefect 2.0

**Reply by zanieb** [reply]
@anna-geller @WillRaphaelson I'm not sure this is possible yet? We should
1. See if we can use this same pattern of exiting early based on the scheduled time in the state in the context
2. If that is not possible, make it possible
3. Consider a better/first-class pattern

**Reply by anna-geller** [reply]
Created a separate issue for this for v2 https://github.com/PrefectHQ/prefect/issues/7006
